components/sync15/src/client/coll_update.rs

```diff
@@ -43,19 +46,17 @@ pub fn fetch_incoming(
         // That should cause us to re-read crypto/keys and things should
         // work (although if for some reason crypto/keys was updated but
         // not all storage was wiped we are probably screwed.)
-        let decrypted = record.decrypt(&state.key)?;
-        result.changes.push(decrypted.into_timestamped_payload());
+        result.changes.push(record.into_decrypted(&state.key)?);
```

Comment:
I'm loving all the small changes like where the new code is cleaner and more readable.